python/ray/train/constants.py

```diff
@@ -45,6 +45,25 @@ def _get_ray_train_session_dir() -> str:
 # Deprecated configs can use this value to detect if the user has set it.
 _DEPRECATED_VALUE = "DEPRECATED"
 
+
+# ==================================================
+#               Train V2 constants
+# ==================================================
+
+# Set this to 1 to enable deprecation warnings for V2 migration.
+ENABLE_V2_MIGRATION_WARNINGS_ENV_VAR = "RAY_TRAIN_ENABLE_V2_MIGRATION_WARNINGS"
+
+
+V2_MIGRATION_GUIDE_MESSAGE = (
+    "See this issue for more context and migration options: "
+    "https://github.com/ray-project/ray/issues/49454"
+)
+
+
+def _v2_migration_warnings_enabled() -> bool:
+    return env_bool(ENABLE_V2_MIGRATION_WARNINGS_ENV_VAR, False)
+
```

Comment:
Could you include in the PR description when these warning messages will be enabled by default?

And what an expected output looks like.